### PR TITLE
Move from C++11 to C++17 (second try)

### PR DESCRIPTION
**Pull request by julianoes**
Previous try was #884.

### CODE REVIEW

**Comment by JonasVautherin** [comment]
Almost there, just missing iOS. I see a lot of errors of the sort: "only available on iOS 11 or newer":

```
error: aligned allocation function of type 'void *(std::size_t, std::align_val_t)' is only available on iOS 11 or newer
```

**Review by JonasVautherin — DISMISSED** [review]

**Comment by julianoes** [comment]
Rebased on top of develop/496795c1.

File: src/core/mavlink_parameters.h [hunk 1]
```diff
@@ -238,25 +222,25 @@ class MAVLinkParameters {
 
         bool set_as_same_type(const std::string& value_str)
         {
-            if (_value.is<uint8_t>()) {
+            if (std::get_if<uint8_t>(&_value)) {
```
Comment (JonasVautherin) [inline src/core/mavlink_parameters.h:225]: I suppose `_value.is` does not work with a variant, hence this change?

**Review by JonasVautherin — APPROVED** [review]
Nice! :rocket:

File: src/core/mavlink_parameters.h (hunk 1 above)
Comment (julianoes) [reply src/core/mavlink_parameters.h:225]: I think so. There is also `std::get` which throws instead: https://en.cppreference.com/w/cpp/utility/variant/get

**Comment by julianoes** [comment]
:tada:

**Pull request merged**